File: kyu7_2024/ExcelSheetColumnNumbers/ExcelSheetColumnNumbers.kt
// There isn't one for this language, but I practiced logic the same way

//  Excel sheet column numbers

object Kata {
    fun titleToNumber(title: String): Int {
          val alphabetLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
          
          var resultValue = 0;
          for (letter in title) {
            val letterValue = alphabetLetters.indexOf(letter) + 1

            println("$letterValue $letter")
            resultValue = resultValue * 26 + letterValue;
            println(letter)
        }
  
          return resultValue
        
    }
  }

  fun main() {
    val result = Kata.titleToNumber("AA") 
    println(result)
}

File: kyu7_2024/ExcelSheetColumnNumbers/ExcelSheetColumnNumbers.js
// https://www.codewars.com/kata/55ee3ebff71e82a30000006a/train/javascript

// Excel sheet column numbers

function titleToNumber(title) {
    const alphabetLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'    
    
    let resultValue = 0
    for (let i = 0; i < title.length; i++) {
      const currentLetter = title[i]
      const letterValue = alphabetLetters.indexOf(currentLetter) + 1
      
      resultValue = resultValue * 26 + letterValue;
    }
    
    return resultValue
  }